runner_manager/backend/aws.py

```diff
@@ -65,20 +64,17 @@ def get(self, instance_id: str) -> Runner:
             error = e.response.get("Error", {})
             if error.get("Code") == "InvalidInstanceID.NotFound":
                 log.error(f"Instance {instance_id} not found.")
+                return Runner.find(Runner.instance_id == instance_id).first()
```

Comment:
I don't understand this part. If we don't find it, we still return it? What do we return here exactly?

Reply:
so it tries to find the instance and return none if no match is found.

Reply:
Why not returning `None` then? Why this call to `Runner.find`?

Reply:
Something is up indeed, I think this function should retrieve the instance name thanks to the instance_id. And then try to find the runner with the `name` instead of the `instance_id` otherwise there's no point of using it.

Reply:
We could find an base exception that can be raised or create a new one under `runner_manager/exceptions.py`


But for now let's leave as is and raise NotFoundError just for consistency with the base backend. this method is not yet used in the rest of the code and might be deleted at the end.

Reply:
removed the get method.